### ISSUE TITLE
WebRTC support for LOQUI

### ISSUE THREAD

**Issue by Gioyik** [loqui/im#54]
Hi,

I think that LOQUI needs to level up with this feature. I had those useful links to implement it on LOQUI:

https://github.com/ESTOS/strophe.jingle
https://github.com/ESTOS/gowebrtc.js

Is interesting, and the best part is that is working. I read about it and could be possible to integrate to LOQUI, because is a plug in for Strophe.js. Probably you are thinking in how to test WebRTC if the full support in Firefox OS will come with the v1.3, so this is the answer:

https://wiki.mozilla.org/Media/WebRTC/Peak

I have some work for the UI, so tell me if you are interested in add WebRTC support for LOQUI and if is possible for sure, to make a pull request an show you the changes in the UI for the VideoCall support and make the changes in the code of the pull that you think convenient.

Regards,
Gio

**Comment by aesedepece** [comment]
Hi Gioyik.

WebRTC is definitely in the roadmap, and personally I'd want to support it by the end of the year.

The code from ESTOS implementing it as a strophe plugin should make it incredibly easy to add to Loqui. Having the ability to test it in Peak devices right now sounds great too, didn't knew that.

It would be GREAT if you could work in that pull request :D . So, FYI release v0.1 will be released next Tuesday 15th. There are substantial changes to the source code, so you may want to wait for it before starting to code anything.

By the way, "LOQUI" will be re-branded as "Loqui IM" which is much more self-explanatory. Many other changes are yet to come!

Thanks for all your enthusiasm. At Waaltcom we are thinking of ways to work side-by-side with pull requesters and reward you in any way. 

Regards!

**Comment by Gioyik** [comment]
Yes, i was seeing some tweets from Waaltcom twitter account that mention a new release with cool changes. I will wait the new version to help with code in the UI, and if is possible in the way to integrate WebRTC completely. 

I think that is more than possible to have WebRTC support complete before the end of the year. I am positive on that, and i will help to make it possible. :D 

Thanks so much to add me to Waaltcom github organization, I really appreciate it. I will be waiting next release to send pull requests with some code changes.

Cheers!

PD: I think that after the launch of the v 0.1.0 we can have two branches, master and development. What do you think?

**Comment by aesedepece** [comment]
@Gioyik I've assigned this issue to you because Github does not allow more than one assignee per issue...

Let's go for it! I'm creating a development branch where we can work on supporting WebRTC.

**Comment by Gioyik** [comment]
@aesedepece Yup no problem and don't worry about it :+1: 

Thanks for the branch i will wait it to start sending pull requests. Btw, good work with the new version of Loqui, looks great!

Do you have a Peak? Did you try to make a build as explain in the guide/link that i shared you to make possible run WebRTC? I ask just to know, i don't have a build in this moment, i need to build it again. if not, i can said to a friend that could help us to make a build for this :)

**Comment by Gioyik** [comment]
Today a made a push to devel branch (208b0a6ada7f950a9943861ce8f1107c8d3e1201). I will add here every commit that i push related to this topic to have a tracking.

**Comment by Gioyik** [comment]
Yup! 3cf44199a0554fe492e32c4aaedfd09ee7d7229f

**Comment by Gioyik** [comment]
@aesedepece We can start testing the WebRTC support with this library (https://gowebrtc.me/latest.js). Home page (https://gowebrtc.me/) and the demo (https://gowebrtc.me/demo.html).

I think that is necessary to have installed XEP's 0166, 0167, 0293, 0294 and 0320, can you confirm me this? If yes, is possible to install it?

This is a demo that uses strophe.jingle library (http://legastero.github.io/jingle-interop-demos/jsjac-jingle/), we can use it to check if XEP's are installed and running correctly in Loqui server.
